homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -421,19 +568,36 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 
 
 class XiaomiGenericSensor(XiaomiCoordinatedMiioEntity, SensorEntity):
-    """Representation of a Xiaomi Humidifier sensor."""
-
-    def __init__(self, name, device, entry, unique_id, coordinator, description):
+    """Representation of a Xiaomi generic sensor."""
+
+    def __init__(
+        self,
+        name,
+        device,
+        entry,
+        unique_id,
+        coordinator,
+        description: XiaomiMiioSensorDescription,
+    ):
         """Initialize the entity."""
         super().__init__(name, device, entry, unique_id, coordinator)
-
+        self._attr_device_class = description.device_class
+        self._attr_state_class = description.state_class
+        self._attr_icon = description.icon
         self._attr_name = name
         self._attr_unique_id = unique_id
-        self.entity_description = description
+        self._attr_entity_registry_enabled_default = False
```

Comment:
I agree and disagree.

Because this is a `XiaomiGenericSensor` it should be disabled by default. We can either:
1. extend classes from this one and set it to true
2. make it configurable via the description

Registring all generic sensors blindly is not nice and fills user's instances with a lot of sensors that they might never use.

Reply:
Agreed, we should parametrize it for the generic sensor / description, and make it default to `False`. I'd still like to have things like "last clean time" and "last clean area" enabled per default, as that information is likely to be used by most of the people.